thirdparty/zlib/CMakeLists.txt

```diff
@@ -0,0 +1,67 @@
+PROJECT(zlib)
+
+if(NOT WIN32)
+            set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-all")
+            set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-all")
+endif()
+
+if (MSVC)
+  # Build with multiple processes
+  add_definitions(/MP)
+  # MSVC warning suppressions
+  add_definitions(
+    /wd4996 # The compiler encountered a deprecated declaration.
+  )
```

Comment:
1. convert them to `CMAKE_CXX_FLAGS`
2. `if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")`

Reply:
1. Was converted add_definitions  to CMAKE_CXX_FLAGS
2. Condition was changed.